### ISSUE TITLE
Hidden mutation of the native Product model

### ISSUE THREAD

**Issue by akadko** [adaptyteam/AdaptySDK-React-Native#1]
Hello Adapty team,

Thanks for React Native SDK. I found that this wrapper implicitly mutates native Product object model: the 'localizedPrice' field is replaced by the 'price' field's value and the latter is removed.

```
if (product.hasOwnProperty('price')) {
    product.localizedPrice = product.price;
    delete product.price;
}
```

[Here](https://github.com/adaptyteam/AdaptySDK-React-Native/blob/master/lib/src/sdk/paywall.ts#L48) and [here](https://github.com/adaptyteam/AdaptySDK-React-Native/blob/master/lib/src/sdk/paywall.ts#L60).

Could you please explain why is this mutation needed and is there a possibility to retrieve the original 'localizedPrice' value? Thanks in advance.

**Comment by Betra** [comment]
Hey, @akadko!
Mutation is there to unify user experience. You see, Android Billing Client SDK and Apple StoreKit have different naming in their output (`localizedPrice` is an example of StoreKit naming). Mutation you have noticed occurs only for Android users to unite fields. 
We've decided to favour iOS naming as it seems more recognised and popular

If it still doubts you, your opinion is welcomed!

**Comment by akadko** [comment]
@Betra Thanks for a quick response but I still don't get it. 

Why do you consider naming of Billing SDK and StoreKit if this library works as a proxy for Adapty SDK? As docs say, Adapty Product model has the `localizedPrice` string field and the `price` decimal field - both on [iOS](https://docs.adapty.io/sdk/integrating-adapty-sdk/ios-sdk-intro/ios-sdk-models#productmodel) and [Android](https://docs.adapty.io/sdk/integrating-adapty-sdk/android-sdk-intro/android-sdk-sdk-models#productmodel). But RN SDK changes this model unexpectedly.

**Comment by Betra** [comment]
You're absolutely right! I've discovered Android Native SDK has added `localizedPrice` month or two ago. 

We are preparing a big RN SDK later this week, without a doubt this would be fixed!

**Comment by Betra** [comment]
@akadko Sorry to make you wait!
You may find your issue fixed in v0.5 ([Release notes](https://github.com/adaptyteam/AdaptySDK-React-Native/releases/tag/v0.5)) among many others!

**Comment by akadko** [comment]
@Betra Good, thank you!
